doc/configuration.rst

```diff
@@ -99,6 +99,33 @@ Subclassing ``Config`` or ``DefaultConfig``
 
     Defaults to ``False``.
 
+  .. method:: cli_context(command: str) -> Iterator[None]
+
+    A context manager which wraps the execution of the CLI command. Use it to
+    perform any initialization or cleanup required by your codebase.
+
+    ``command`` is the command passed to the monkeytype cli: ``'run'``,
+    ``'apply'``, etc.
+
+    Since :meth:`cli_context` is used as a context manager, any :class:`Config`
```

Comment:
Using `contextlib.contextmanager` will usually be the simplest way to implement this method, but it's not the only way. You can just as well create a separate context manager class (one with `__enter__` and `__exit__` methods) and return an instance of that class from the method. I appreciate the concern for helping people get this right, but I think right now this documentation is a bit too verbose; goes a little too far in documenting the behavior of `contextlib.contextmanager`, which is out of MonkeyType's scope. I would defer to the Python documentation when we can, rely heavily on the example code (which is the first thing most people will look at anyway) and revise to something shorter:

"This method must return a [context manager](https://docs.python.org/3/reference/datamodel.html#with-statement-context-managers) instance. In most cases the simplest way to do this will be via the [contextlib.contextmanager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager) decorator. For example, if you run MonkeyType against a Django codebase, you can setup Django before the command runs::"